Source/Scene/Batched3DModel3DTileContentProvider.js

```diff
@@ -122,75 +118,95 @@ define([
+            that.initialize(arrayBuffer);
+        }).otherwise(function(error) {
             that.state = Cesium3DTileContentState.FAILED;
             that.readyPromise.reject(error);
+        });
+    };
+
+    /**
+     * DOC_TBA
+     */
+    Batched3DModel3DTileContentProvider.prototype.initialize = function(arrayBuffer, byteOffset) {
+        byteOffset = defaultValue(byteOffset, 0);
+
+        var uint8Array = new Uint8Array(arrayBuffer);
+        var magic = getMagic(uint8Array, byteOffset);
+        if (magic !== 'b3dm') {
+            throw new DeveloperError('Invalid Batched 3D Model.  Expected magic=b3dm.  Read magic=' + magic);
         }
 
-        loadArrayBuffer(this._url). then(function(arrayBuffer) {
-            var uint8Array = new Uint8Array(arrayBuffer);
-            var magic = getStringFromTypedArray(getSubarray(uint8Array, 0, Math.min(4, uint8Array.length)));
-            if (magic !== 'b3dm') {
-                throw new DeveloperError('Invalid Batched 3D Model.  Expected magic=b3dm.  Read magic=' + magic);
-            }
+        var view = new DataView(arrayBuffer);
+        byteOffset += sizeOfUint32;  // Skip magic number
 
-            var view = new DataView(arrayBuffer);
-            var byteOffset = 0;
+        //>>includeStart('debug', pragmas.debug);
+        var version = view.getUint32(byteOffset, true);
+        if (version !== 1) {
+            throw new DeveloperError('Only Batched 3D Model version 1 is supported.  Version ' + version + ' is not.');
+        }
+        //>>includeEnd('debug');
+        byteOffset += sizeOfUint32;
 
-            byteOffset += sizeOfUint32;  // Skip magic number
+        // Skip byteLength
+        byteOffset += sizeOfUint32;
 
-            //>>includeStart('debug', pragmas.debug);
-            var version = view.getUint32(byteOffset, true);
-            if (version !== 1) {
-                throw new DeveloperError('Only Batched 3D Model version 1 is supported.  Version ' + version + ' is not.');
-            }
-            //>>includeEnd('debug');
-            byteOffset += sizeOfUint32;
-
-            var batchTableLength = view.getUint32(byteOffset, true);
-            byteOffset += sizeOfUint32;
-            if (batchTableLength > 0) {
-                var batchTableString = getStringFromTypedArray(getSubarray(uint8Array, byteOffset, batchTableLength));
-                byteOffset += batchTableLength;
-
-                // PERFORMANCE_IDEA: is it possible to allocate this on-demand?  Perhaps keep the
-                // arraybuffer/string compressed in memory and then decompress it when it is first accessed.
-                //
-                // We could also make another request for it, but that would make the property set/get
-                // API async, and would double the number of numbers in some cases.
-                batchTableResources.batchTable = JSON.parse(batchTableString);
-            }
+        // TODO : rename to batchLength?
```

Comment:
Then it would be zero.  This is consistent with 3D Tiles naming: https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/26

What do you suggest?